.github/workflows/test-upstream.yml

```diff
@@ -44,6 +45,12 @@ jobs:
         with:
           toolchain: stable
           override: true
+      - name: Update upstream cargo dependencies
+        run: |
+          cd dask_planner
+          mv test-upstream.toml Cargo.toml
```

Comment:
Rather than maintain a separate Cargo.toml, could we run a script here to rewrite the relevant imports?

Reply:
I'm in favor of something like that as well. I played around a bit with diff/git merge-files to rewrite the the arrow datafusion lines but my bash foo was failing me. 

Do you know offhand any commands that could be useful? If not, I can explore this a bit more (or maybe just do it in a python script)

Reply:
```bash
$ sed -i -r 's/^datafusion-([a-z]+).*/datafusion-\1 { git = "https:\/\/github.com\/apache\/arrow-datafusion\/" }/g' Cargo.toml
```

```bash
$ git diff
diff --git a/dask_planner/Cargo.toml b/dask_planner/Cargo.toml
index 51826ab..1f7a686 100644
--- a/dask_planner/Cargo.toml
+++ b/dask_planner/Cargo.toml
@@ -11,10 +11,10 @@ rust-version = "1.62"
 [dependencies]
 arrow = { version = "23.0.0", features = ["prettyprint"] }
 async-trait = "0.1.41"
-datafusion-common = { git = "https://github.com/apache/arrow-datafusion/", rev = "1261741af2a5e142fa0c7916e759859cc18ea59a" }
-datafusion-expr = { git = "https://github.com/apache/arrow-datafusion/", rev = "1261741af2a5e142fa0c7916e759859cc18ea59a" }
-datafusion-optimizer = { git = "https://github.com/apache/arrow-datafusion/", rev = "1261741af2a5e142fa0c7916e759859cc18ea59a" }
-datafusion-sql = { git = "https://github.com/apache/arrow-datafusion/", rev = "1261741af2a5e142fa0c7916e759859cc18ea59a" }
+datafusion-common { git = "https://github.com/apache/arrow-datafusion/" }
+datafusion-expr { git = "https://github.com/apache/arrow-datafusion/" }
+datafusion-optimizer { git = "https://github.com/apache/arrow-datafusion/" }
+datafusion-sql { git = "https://github.com/apache/arrow-datafusion/" }
 env_logger = "0.9"
 log = "^0.4"
 mimalloc = { version = "*", default-features = false }

```

Reply:
Thanks for the suggestion. That did the trick!